packages/pwa-kit-dev/src/configs/webpack/plugins/overrides-resolver-plugin.test.js

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {BASE_DIR, runWebpackCompiler} from '../test-utils'
+import OverridesResolverPlugin from './overrides-resolver-plugin'
+
+describe('Overrides Resolver Plugin', () => {
+    const testCases = [
+        {
+            description: 'Wildcard import resolved to correct extension override',
+            entryPoint: 'app/routes.jsx',
+            // Compiler configuration.
+            compilerConfig: {
+                extensions: ['@salesforce/extension-this', '@salesforce/extension-that'],
+                files: {
+                    // Virtual Project Files
+
+                    // Entry Point
+                    '/virtual/project/app/routes.jsx': `import SamplePage from '*/pages/sample-page'`,
+
+                    // Overrides
+
+                    // Local Files
+                    '/virtual/project/app/pages/sample-page.jsx': '// Sample Page',
+
+                    // Extensions Overrides
+                    '/virtual/project/node_modules/@salesforce/extension-that/src/overrides/pages/sample-page.jsx':
+                        '// @salesforce/extension-that',
+                    '/virtual/project/node_modules/@salesforce/extension-this/src/overrides/pages/sample-page.jsx':
+                        '// @salesforce/extension-this'
```

Comment:
The tests in this file have the app extensions overriding the sample page that is defined in the base project. Would it be better if it's the other way around? 

The base project overriding files in the extensions makes sense. But the extensions themselves should not have any prior knowledge of what's in the base project.